source/Halibut/Transport/TcpClientManager.cs

```diff
@@ -62,22 +62,25 @@ public void RemoveClient(TcpClient client)
         {
             lock (syncLock)
             {
-                foreach(var thumbprintClientsPair in activeClients)
+                if (activeClients != null)
```

Comment:
This was throwing during debug of some tests

Reply:
Although, why is it null? It looks like it's only null because it was disposed. 

Is it an issue that our tests are causing us to access an object that has been disposed? 🤔 

If we're cool with that, an alternative is to set `activeClients` to a new empty object if we didn't want to do a null check here. 

But it still feels a little sus that we're accessing a disposed object.